File: projects/Занько/yana3/app/src/main/java/com/example/lr3/ui/detail/PostDetailFragment.java
package com.example.lr3.ui.detail;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.example.lr3.databinding.FragmentPostDetailBinding;
import com.example.lr3.model.Post;

public class PostDetailFragment extends Fragment {

    private FragmentPostDetailBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentPostDetailBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        Post post = null;
        if (getArguments() != null) {
            Object obj = getArguments().getSerializable("post");
            if (obj instanceof Post) post = (Post) obj;
        }

        if (post != null) {
            binding.tvDetailId.setText("ID: " + post.getId() + " • userId: " + post.getUserId());
            binding.tvDetailTitle.setText(post.getTitle());
            binding.tvDetailBody.setText(post.getBody());
        } else {
            binding.tvDetailTitle.setText("Нет данных");
            binding.tvDetailBody.setText("Элемент не был передан.");
        }

        // Явная кнопка назад (плюс будет Up-кнопка в тулбаре)
        binding.btnBack.setOnClickListener(v ->
                NavHostFragment.findNavController(this).navigateUp()
        );
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: projects/Ярома/LAB3/app/src/main/java/com/example/lab3/PostDetailFragment.kt
package com.example.lab3

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.lab3.databinding.FragmentPostDetailBinding

class PostDetailFragment : Fragment() {

    private var _binding: FragmentPostDetailBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentPostDetailBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val id = arguments?.getInt("post_id", 0) ?: 0
        val userId = arguments?.getInt("post_userId", 0) ?: 0

        binding.tvAuthor.text = "Цеван Константин АС-66"
        binding.tvId.text = "Номер записи: $id"
        binding.tvUserId.text = "Пользователь: $userId"
        binding.tvTitle.text = "Подробная информация по записи"
        binding.tvBody.text = buildFullDescription(id, userId)

        binding.btnBack.setOnClickListener {
            requireActivity().finish()
        }
    }

    private fun buildFullDescription(id: Int, userId: Int): String {
        return "Описание: данная запись была загружена с удаленного сервера в формате JSON " +
                "и преобразована в объект приложения.\n\n" +
                "Элемент списка имеет номер $id и относится к пользователю $userId.\n\n" +
                "На главном экране запись отображается в виде карточки, а при выборе открывается " +
                "отдельный экран с детальной информацией внутри фрагмента."
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}